lib/Doctrine/CouchDB/HTTP/SocketClient.php

```diff
@@ -290,12 +290,10 @@ public function request($method, $path, $data = null, $raw = false, array $heade
                     while ($bytesLeft > 0) {
                         $body .= $read = fread($this->connection, $bytesLeft + 2);
                         $bytesLeft -= strlen($read);
+                        $body = substr($body, 0, -2);
                     }
                 }
             } while ($bytesToRead > 0);
-
-            // Chop off \r\n from the end.
-            $body = substr($body, 0, -2);
```

Comment:
We already removed that symbols in while loop, and there we just make JSON become invalid again.